CRM/PivotData/AbstractData.php

```diff
@@ -177,6 +187,51 @@ protected function replaceCustomLabels($label) {
     return $label;
   }
 
+  /**
+   * Loads data for the given custom field.
+   *
+   * @param array $field
+   *   Array with the field's data, with the field's name and the field's group
+   *
+   * @return array
+   */
+  protected function loadCustomFieldData($field) {
+    if (empty($field['name']) || empty($field['group'])) {
+      return array();
+    }
+
+    $sql = '
+      SELECT g.id AS group_id, f.id AS id, f.label AS label, f.data_type AS data_type, 
+             f.html_type AS html_type, f.date_format AS date_format, 
+             og.name AS option_group_name 
+        FROM `civicrm_custom_group` g 
+   LEFT JOIN `civicrm_custom_field` f ON f.custom_group_id = g.id 
+   LEFT JOIN `civicrm_option_group` og ON og.id = f.option_group_id 
+       WHERE f.name = %1
+         AND g.name = %2
+    ';
+    $customFieldsResult = CRM_Core_DAO::executeQuery($sql, array(
+      1 => array($field['name'], 'String'),
+      2 => array($field['group'], 'String'),
+    ));
+    $customFieldsResult->fetch();
+
+    $customField = new CRM_Core_BAO_CustomField();
+    $customField->id = $customFieldsResult->id;
+    $customField->find(true);
+
+    $field = array(
+      'name' => 'custom_' . $customFieldsResult->id,
+      'title' => $customFieldsResult->label,
+      'pseudoconstant' => array(
+        'optionGroupName' => $customFieldsResult->option_group_name,
```

Comment:
what if the custom field does not have an option group (like a text field) ?

Reply:
No matter, that is taken care of by the left join, it'll just be empty and this is handled later in the execution. This is how other custom fields are being loaded for other entities, so I thought I better leave everything consistent.

Eg. 
https://github.com/compucorp/uk.co.compucorp.civicrm.pivotreport/blob/2ff5e1b39e5471edb08fbbb5a56601aac039e493/CRM/PivotData/DataProspect.php#L111-L133